app/js/components/manageApps/AddAddon.jsx

```diff
@@ -6,68 +6,64 @@
  * graphic logo is a trademark of OpenMRS Inc.
  */
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import Dropzone from 'react-dropzone';
 
-export default function AddAddon({
-  handleClear, 
-  handleUpload, 
-  displayManageOwaButtons, 
-  displayManageOwaButtonsState }) {
-  const classesToUse = displayManageOwaButtonsState ?
-    'col-xs-6 col-sm-8' : 'col-xs-6 col-sm-12';
-  return (
-    <div className="container-fluid">
-      <fieldset className="scheduler-border">
-        <legend className="scheduler-border">Upload Addon Package:</legend>
-        <div className="row">
-          <div className="col-xs-12">
-            <div className={classesToUse}>
-              <input
-                name="file"
-                type="file" 
-                id="fileInput"
-                onChange={displayManageOwaButtons}
-                accept=".zip"
-                data-text="Browse Addon"
-                data-buttonName="btn-primary"
-                data-placeholder="Select Addon to upload"
-                data-btnClass="btn-primary"
-                className="filestyle" />
-            </div>
-            {
-              displayManageOwaButtonsState && 
-                <div className="col-xs-6 col-sm-4">
-                  <div className="row">
-                    <div className="btn-toolbar">
-                      <button
-                        className="btn btn-primary"
-                        onClick={handleUpload}>
-                        <span
-                          className="glyphicon glyphicon-upload"
-                        /> Upload
-                      </button>
-                      <button
-                        id="cancel-btn"
-                        className="btn btn-danger"
-                        onClick={handleClear}
-                      >
-                        <span className="glyphicon glyphicon-remove" /> Clear
-                      </button>
-                    </div>
-                  </div>
-                </div>
-            }
-          </div>
+export default class AddAddon extends Component {
+  constructor() {
+    super();
+    this.state = {
+      disableClick: true,
+      dropzoneRef: {},
+    }
+    this.showMessage = this.showMessage.bind(this);
+  }
+
+  showMessage() {
+    if (this.props.files == null || this.props.files.length < 1) {
+      return (
+        <p>Drag file here to upload or
+          <a id="click-to-select" href=""
+            onClick={(e) => {
+              e.preventDefault(); this.state.dropzoneRef.open();
+            }}> Click to here select</a>
+        </p>
+      );
+    }
+    else {
+      return (
+        <p>{this.props.files[0].name}</p>
+      );
+    }
+  }
+
+  render() {
+    return (
+      <div className="container-fluid">
+        <div className="row drop-zone-container">
+          <span>Select an OWA to upload</span>
```

Comment:
I think this span can be removed, we just need to change the wording on the file upload field to be more descriptive e.g `Drag addon file(s) here to upload or Click to here select`

Reply:
@andela-jomadoye Thanks for pointing out the clearing of the drag and drop field on successful upload. I'll work on that.

On the span, I think the drag and drop field should have a title/label/description other than the instruction for a call to action. Maybe I should look for better wording?